packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt

```diff
@@ -200,6 +215,84 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
         isRegistered.set(false)
     }
 
+    @Suppress("TooGenericExceptionCaught")
+    private fun mapLogEvent(event: LogEvent): LogEvent? {
+        val jsonEvent = event.toJson().asMap()
+
+        var modifiedJson: Map<String, Any?>? = null
+
+        val semaphore = Semaphore(1)
+        semaphore.acquire()
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                channel.invokeMethod(
+                    "mapLogEvent",
+                    mapOf(
+                        "event" to jsonEvent
+                    ),
+                    object : MethodChannel.Result {
+                        override fun success(result: Any?) {
+                            modifiedJson = result as? Map<String, Any?>
+                            semaphore.release()
+                        }
+
+                        override fun error(
+                            errorCode: String,
+                            errorMessage: String?,
+                            errorDetails: Any?
+                        ) {
+                            // No telemetry needed, this is likely an issue in user code
+                            semaphore.release()
+                        }
+
+                        override fun notImplemented() {
+                            Datadog._internal._telemetry.error(
+                                "mapLogEvent returned notImplemented."
+                            )
+                            semaphore.release()
+                        }
+                    }
+                )
+            } catch (t: Throwable) {
+                Datadog._internal._telemetry.error("Attempting call mapLogEvent failed.", t)
+                semaphore.release()
+            }
+        }
+        // Stalls until the method channel finishes
+        semaphore.acquire()
```

Comment:
alternatively we can use `CountdownLatch(1)` which has `await` method which can take a timeout (to avoid cases waiting forever, if that can happen) and will reduce 2 `acquire` calls to the single `await` here (and in the `post` we need to call only `countDown`).

Reply:
👍 I'll look into it.